### ISSUE TITLE
[css-text-3] ::first-letter is typographically misplaced

### ISSUE THREAD

**Issue by hunboy** [w3c/csswg-drafts#965]
Currently the ::first-letter is typographically misplaced. To solve this issue some spec declaration would be useful.

I collected the issue at the mozbugzilla with samples, explanations, workarounds, troubleshoots, etc during the years.

Please, visit for discussion:
https://bugzilla.mozilla.org/show_bug.cgi?id=415506

Summary applied on ::first-letter pseudo:

 -moz-versal-type: raised|dropped|adjacent|none
 -moz-versal-egality: kerning|none
 -moz-versal-lines: {decimal number}
 
shorthand sample is:

 -moz-versal: dropped kerning 5.5;

default is: none none 1

**Comment by tabatkins** [comment]
Floated ::first-letter layout allows browsers to do "smart" drop-cap layouts, but doesn't define how it works.

Are you trying to suggest new properties equivalent to the ['initial-letter' properties](https://drafts.csswg.org/css-inline-3/#propdef-initial-letter)?

**Comment by hunboy** [comment]
Seems, yeah. I missed this new spec, now analysing.
Only a few comment is: 

- initlial-letter: 3 inifinite (or something)
  should be the adjacent initial .

- the initial-letter: 2.51 3; sample is typograhically invalid, however someone will like that.
  Reason: The dropped initiale cap-height must reach the first line's cap-height, or
   can be above of that, but never below.

summary: this is good for raised and dropped and allows sunken but doesn't allow adjacent currently.

I like the new spec. So I don't want to rename them, I just copied here the ideacollection we made years ago.

**Comment by hunboy** [comment]
Well yes, this issue is almost complete except adjacent initial-letter, so I modify the title for that.

Workaround is here for adjacent sample: https://jsfiddle.net/utasir/vgfz13L1/

**Comment by hunboy** [comment]
Note, there are some issues with the adjacent initiale:

 - adjacent initiale for paged media is applied for all paragraphs exclusively on the first page, where it is defined.
 - adjacent initiale stops at the forwarding pages.

These issues are for the paged media.

**Comment by hunboy** [comment]
![gutenberg](https://cloud.githubusercontent.com/assets/547294/22272165/c4aebe42-e298-11e6-9cf3-a90727002bc9.jpg)

The first Gutenberg Bible with an adjacent cap (marked as red)

**Comment by hunboy** [comment]
![image12_w650](https://cloud.githubusercontent.com/assets/547294/22272485/a5c07942-e29a-11e6-91b2-81e90c9a949e.png)

The three common initiale in the hungarian typography (source: University of ELTE Tipography lessions)
süllyesztett (dropped), lógatott (adjacent) és kiemelt (raised)

**Comment by hunboy** [comment]
http://theworldsgreatestbook.com/book-design-part-6/

International sample (typography knowledge base) "Hanging" Cap

**Comment by hunboy** [comment]
![hanging](https://cloud.githubusercontent.com/assets/547294/22342131/8c25bfd8-e3f3-11e6-95cd-4c88aecd32d4.jpg)

A scan from a book: Péter Virágvölgyi - The profession of typography

**Comment by fantasai** [comment]
We have an outstanding WG resolution to make width/height apply to initial letter. Once you have that, this is easy: set the first-letter to a width, then give it a negative margin-start of that same width. This will place it in the margin.

https://github.com/w3c/csswg-drafts/issues/863

**Comment by fantasai** [comment]
OK, we've applied width/height to the initial letter in #863, so hanging the initial letter in the margin should be possible now.

@hunboy We can certainly do the initial letter in the margin, as illustrated in your manuscript. But it's not entirely clear what's happening in the magazine clip. Can you clarify what’s expected to happen if

- If there is a another paragraph after the one with the hanging initial letter? Is that paragraph also indented, or does it start underneath the hanging initial letter?

- If the first paragraph is short enough that the second paragraph begins halfway between the top and bottom of the hanging initial letter?

We need to know if hanging in the margin is sufficient, or if we need a different form of clearance for `initial-letter-wrap`, and if so how exactly that operates.

**Comment by hunboy** [comment]
@fantasai the adjacent initial letter introduces at the paged media 500 years ago so we need fix, if what this means when we are on a non paged media, probably margin reaches the end of the document, not only the current paragraph. On paged media it's clear: margin stays ONLY on the current page. (the whole page, not only the current paragraph). So shortly, the margin is wider, and the place for cap is the margin.

Here is a sample what I'm talking about:

http://jsfiddle.net/utasir/34nkw9ab/

This explains lots of your questions.

**Comment by hunboy** [comment]
@fantasai here is a two column sample:

http://jsfiddle.net/utasir/34nkw9ab/1/

only the first column is affected by the margin, the second one not.

**Comment by AmeliaBR** [comment]
If the goal is to have a margin that auto-computes based on the initial size, and only applies to the current layout fragment (page and/or column), then maybe this would be best described as a new value of the [`-wrap` property](https://drafts.csswg.org/css-inline-3/#initial-letter-wrapping)?

**Comment by fantasai** [comment]
@AmeliaBR Yes, it would have to be a new value for that property, one that extends to the next initial letter, the end of the block formatting context, or the bottom of the fragmentainer, whichever comes first.

Flagging this for the next level, I think there's more than enough complexity for a first level in the current spec. :/

**Comment by fantasai** [comment]
We'll add an example of how to do the basic case of putting the initial letter in a gutter to the L3 spec, though; that should handle the most common situations, and can be done just with width + negative margin.

**Comment by dauwhe** [comment]
```css
p::first-letter {
  initial-letters: 3;
  color: red;
  width: 72pt;
  text-align: right;
  margin-left: -72pt;
}

p {
  margin-left: 72pt;
}
```

<img width="601" alt="adjacent-initial-letter" src="https://user-images.githubusercontent.com/5687700/47113375-e0fbf900-d226-11e8-80f7-cf25e2d701ee.png">
